RatScanner/ScreenInfo.cs

```diff
@@ -0,0 +1,213 @@
+			public RECT rcNormalPosition;
+
+			public override string ToString()
+			{
+				byte[] StructBytes = RawSerialize(this);
+				return System.Convert.ToBase64String(StructBytes);
+			}
+
+			public void ReadFromBase64String(string aB64)
+			{
+				byte[] b64 = System.Convert.FromBase64String(aB64);
+				var NewWP = ReadStruct<WINDOWPLACEMENT>(b64, 0);
+				length = NewWP.length;
+				flags = NewWP.flags;
+				showCmd = NewWP.showCmd;
+				ptMinPosition.x = NewWP.ptMinPosition.x;
+				ptMinPosition.y = NewWP.ptMinPosition.y;
+				ptMaxPosition.x = NewWP.ptMaxPosition.x;
+				ptMaxPosition.y = NewWP.ptMaxPosition.y;
+				rcNormalPosition.left = NewWP.rcNormalPosition.left;
+				rcNormalPosition.top = NewWP.rcNormalPosition.top;
+				rcNormalPosition.right = NewWP.rcNormalPosition.right;
+				rcNormalPosition.bottom = NewWP.rcNormalPosition.bottom;
+			}
+
+			static public T ReadStruct<T>(byte[] aSrcBuffer, int aOffset)
+			{
+				byte[] buffer = new byte[Marshal.SizeOf(typeof(T))];
+				Buffer.BlockCopy(aSrcBuffer, aOffset, buffer, 0, Marshal.SizeOf(typeof(T)));
+				GCHandle handle = GCHandle.Alloc(buffer, GCHandleType.Pinned);
+				T temp = (T)Marshal.PtrToStructure(handle.AddrOfPinnedObject(), typeof(T));
+				handle.Free();
+				return temp;
+			}
+
+			static public T ReadStruct<T>(Stream fs)
+			{
+				byte[] buffer = new byte[Marshal.SizeOf(typeof(T))];
+				fs.Read(buffer, 0, Marshal.SizeOf(typeof(T)));
+				GCHandle handle = GCHandle.Alloc(buffer, GCHandleType.Pinned);
+				T temp = (T)Marshal.PtrToStructure(handle.AddrOfPinnedObject(), typeof(T));
+				handle.Free();
+				return temp;
+			}
+
+			public static byte[] RawSerialize(object anything)
+			{
+				int rawsize = Marshal.SizeOf(anything);
+				byte[] rawdata = new byte[rawsize];
+				GCHandle handle = GCHandle.Alloc(rawdata, GCHandleType.Pinned);
+				Marshal.StructureToPtr(anything, handle.AddrOfPinnedObject(), false);
+				handle.Free();
+				return rawdata;
+			}
+		}
+
+		public static List<ScreenScale> GetScreenScales()
+		{
+			List<ScreenScale> scales = new List<ScreenScale>();
+			// Look through all screens and calculate the scaling on each screen and return that as a screenscale with bounds
+			foreach(Screen screen in Screen.AllScreens)
+			{
+
+				double scale = GetScalingForPoint(new System.Drawing.Point(screen.Bounds.X, screen.Bounds.Y));
+
+				scales.Add(new ScreenScale(scale, screen.Bounds));
+			}
+			return scales;
+		}
+
+		public static ScreenScale? GameWindowScreenScale()
+		{
+			try
+			{
+				System.Drawing.Rectangle gameRect = GameWindowLocator.GetWindowLocation();
+				// Find the ScreenScale where the bounds of the EFT game window are within the bounds of the ScreenScale
+				// The gameRect is screwed up. The Width is actually the right hand bound, and the Height is the lower bound
+				// So we have tod so some special checks instead of a .Contains(gameRect) call.
+				return GetScreenScales().Where(s => s.bounds.Contains(gameRect)).First();
```

Comment:
The important bit which finds the monitor with the game in it. In theory if the game spanned multiple displays, this might break? I didn't test that case, but I don't think we need to necessarily support a weird case like that (I believe something like Nvidia Surround would work since its technically one display to Windows)